waiter/src/waiter/scheduler/shell.clj

```diff
@@ -607,6 +641,34 @@
     (let [id->service @id->service-agent]
       (map (fn [[_ {:keys [service]}]] service) id->service)))
 
+  (signal-instance [this {:keys [id service-id] :as instance} signal-type] 
+   (log/info "in signal-instance")
+   (if (scheduler/service-exists? this service-id)
+      (let [completion-promise (promise)
+            message "Killed using scheduler API"]
+        (send id->service-agent kill-instance service-id id message
+               port->reservation-atom port-grace-period-ms
+               completion-promise)
+        (let [result (deref completion-promise) ;;(signal-instance
+                     ;;service-id id message
+                     ;;port->reservation-atom port-grace-period-ms
+                     ;;completion-promise)
+              success (= result :deleted)]
+          (when success
+            (scheduler/log-service-instance instance :kill :info))
+          {:killed? true
+           :success success
+           :result result
+           :message (if success
```

Comment:
As we avoid blocking, message should read that we have acknowledged request to send sigkill to instance.